scripts/build/types.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package build
+
+import (
+	"os"
+)
+
+// Config contains various pieces of configuration created by main and passed
+// around.
+type Config struct {
+	ArchiveFormat string
+	BuildFormats  []string
+	OutputPath    string
+	Packages      []string
+	Run           bool
+	Verbose       bool
+}
+
+// Intermediate files are stored in memory so you do not need sudo to create
+// cpio files. Some fields such as atime and mtime are not included here
+// because they are useless for our purposes.
+type file struct {
```

Comment:
I haven't considered it. Likely, it would be the case that the program swaps and OOMs.

However, it can be fixed with a small tweak. You can mmap a file to a `byte[]`.

If that is too adventurous, `data []byte` can be replace by a `struct{originalPath string; data []byte}`. If `originalPath` is `""`, then the file's data is in `data`, essentially reimplementing mmap functionality.

The third optional is to never keep file contents in memory and just have an `originalPath`.

What's currently in the PR is good enough for now. When the need arises later, there a few options.